SmartDeviceLinkTests/DevAPISpecs/SDLLifecycleManagerSpec.m

```diff
@@ -430,10 +429,12 @@ @interface SDLLifecycleManager ()
                     registerAppInterfaceResponse.resultCode = SDLResultWrongLanguage;
                     registerAppInterfaceResponse.info = @"Language mismatch";
                     registerAppInterfaceResponse.language = SDLLanguageEnGb;
+                    registerAppInterfaceResponse.hmiDisplayLanguage = SDLLanguageEnGb;
                     testManager.registerResponse = registerAppInterfaceResponse;
 
                     SDLLifecycleConfigurationUpdate *update = [[SDLLifecycleConfigurationUpdate alloc] initWithAppName:@"EnGb" shortAppName:@"E" ttsName:[SDLTTSChunk textChunksFromString:@"EnGb ttsName"] voiceRecognitionCommandNames:nil];
                     OCMStub([testManager.delegate managerShouldUpdateLifecycleToLanguage:[OCMArg any]]).andReturn(update);
+                    OCMStub([testManager.delegate managerShouldUpdateLifecycleToLanguage:[OCMArg any] hmiLanguage:[OCMArg any]]).andReturn(update);
```

Comment:
Given the above change request to only call the old delegate method if it's the only one that's implemented, I don't think this change will work. It's going to require a bigger refactor of the tests to test this properly. We'd need two separate test classes that will implement one or the other method and make sure they're called properly. Because of these difficult requirements I'm okay with it if you remove all changes in the test file and Livio will refactor the tests to account for this case.

I have opened a branch called `feature/issue-1606-sdl-0289-set-language-separately`. Please re-target this PR to that branch so that we can write the tests before merging to `develop`.